pkgs/by-name/ba/baidupcs-go/package.nix

```diff
@@ -0,0 +1,25 @@
+{
+  buildGo122Module,
+  lib,
+  fetchFromGitHub,
+}:
+buildGo122Module rec {
+  pname = "baidupcs-go";
+  version = "3.9.5";
+  src = fetchFromGitHub {
+    owner = "qjfoidnh";
+    repo = "BaiduPCS-Go";
+    rev = "v${version}";
+    hash = "sha256-zNodRQzflOOB3hAeq4KbjRFlHQwknVy+4ucipUcoufY=";
+  };
+
+  vendorHash = "sha256-msTlXtidxLTe3xjxTOWCqx/epFT0XPdwGPantDJUGpc=";
+  doCheck = false;
```

Comment:
Requires network? It may be possible to only disable some of them, see https://nixos.org/manual/nixpkgs/unstable/#ssec-skip-go-tests. Also it would be good to add https://nixos.org/manual/nixpkgs/unstable/#versioncheckhook.

Reply:
Several tests on upstream are just broken and do not build. The compiled binary works fine though.

Reply:
Then please add versionCheckHook so we can make sure it can at least run